autoload/deoplete/handlers.vim

```diff
@@ -51,6 +51,8 @@ function! s:completion_begin(event) abort "{{{
   " Save the previous position
   let g:deoplete#_context.position = context.position
 
+  call deoplete#mappings#_set_completeopt()
```

Comment:
No.  You should not call the function in the position.
The function must be called before of `feedkeys()` or `complete()`.